### ISSUE TITLE
Remove bindings set from evil-collection-magit-mode-map-bindings

### ISSUE THREAD

**Issue by angrybacon** [emacs-evil/evil-collection#714]
Other relevant issue: https://github.com/emacs-evil/evil-collection/issues/214

I have a global binding `gf` to find files

```el
(use-package evil
  :defines evil-normal-state-map
  :bind
  (:map evil-normal-state-map
   ("gf" . me/project-find-file)
   ("gF" . me/project-search))
  :custom
  (evil-want-keybinding nil)
  :hook
  (after-init . evil-mode))
```

So in order to try and remove the `gfp` and `gfu` bindings set here: https://github.com/emacs-evil/evil-collection/blob/88e5ce8bdb0762119b4153ed3346b486029f26d3/modes/magit/evil-collection-magit.el#L354-L355 I tried the following:

```el
(use-package evil-collection
  :functions evil-collection-define-key
  :hook
  (evil-collection-setup . me/evil-collection-reset)
  (evil-mode . evil-collection-init)
  :preface
  (defun me/evil-collection-reset (&rest _rest)
    (with-eval-after-load 'magit
      (unbind-key "<normal-state> gf" magit-status-mode-map)
      (unbind-key "<visual-state> gf" magit-status-mode-map)))
```

Which also disable my own `gf` binding set in the normal map. I've also tried setting the prefix to `nil` for a similar result:

```el
(defun me/evil-collection-reset (&rest _rest)
  (evil-collection-define-key '(normal visual) 'magit-status-mode-map
    "gf" nil))
```

1. I was surprised I couldn't find a way to merge a user-customizable list of bindings into `evil-collection-magit-mode-map-bindings` before the `dolist` kicks in -- although I'm not sure how *removing* items from it would be easily doable. Are there limitations that I'm not seeing?
1. If the above was decided against, what would be the way to exclude bindings set in a mode-specific map (magit-status here) so that they won't collide with global bindings?

**Comment by jojojames** [comment]
Can use the blacklist/whitelist functionality.

```
  (setq evil-collection-key-blacklist '("gfu" "gfp"))
```

**Comment by jojojames** [comment]
Feel free to reopen.

**Comment by angrybacon** [comment]
Oh I didn't about that blacklist feature thank you.

However it doesn't seem to allow the global bindings to take over.

(in my magit-status buffer) <kbd>C-h c</kbd> <kbd>gf</kbd>

> g f is undefined

<kbd>gf</kbd> correctly works in other modes where `evil-collection` doesn't use `gf`

**Comment by jojojames** [comment]
Set those keys to the blacklist if you don't want evil-collection to bind them.

your binding don't seem to be 'global' bindings either which is why other bindings are stepping on them.

Otherwise, I recommend reading below since your issues seem to be user-specific at this point.

https://github.com/emacs-evil/evil-collection#making-spc-work-similarly-to-spacemacs
https://github.com/noctuid/evil-guide

**Comment by angrybacon** [comment]
> Set those keys to the blacklist if you don't want evil-collection to bind them.

That's what I did yeah, which led to `gfp/u` not being bound as expected. But the `gf` binding set for the normal state was not available then. By global I meant not specific to a specific mode map.

I'm guessing that the way I've set bindings to the normal keymap doesn't necessarily mean that 2 keymaps (normal and magit-status) can be active at once and that one keystroke might fallback to one or the other

I'll look at the linked guide thanks
